### PR TITLE
test(definitiontooltip): increase test coverage to 100%

### LINKED ISSUE #17415: Add test coverage for DefinitionTooltip
- [ ] Swarm this one with the whole group
- [ ] document the steps/outcome in the parent issue
- [ ] make issues from the parent list and assign to pairs of 2-3 people
- [ ]

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #17415

#### Changelog

**New**

- added new tests for various functionality that was previously untested

#### Testing / Reviewing

- Codecov should report 100% for DefinitionTooltip
- CI should pass

### CODE REVIEW

**Comment by alisonjoseph** [comment]
Does this mean its 100% now? Screenshot 2024-09-12 at 3 12 51 PM

**Comment by 2nikhiltom** [comment]
How do we check the test coverage percentage locally ?

File: packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js [hunk 1]
```diff
@@ -29,6 +29,123 @@ describe('DefintiionTooltip', () => {
...
+        <DefinitionTooltip
+          data-testid="test"
+          definition={definition}
+          className="tooltip-class"
+          openOnHover>
+          URL
+        </DefinitionTooltip>
+      );
+
+      const content = screen.getByText(definition);
```
Comment (preetibansalui) [inline packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js]: I think we are not using this const. Can we remove it if not using?

File: packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js [hunk 2]
```diff
@@ -29,6 +29,123 @@ describe('DefintiionTooltip', () => {
...
+      render(
+        <DefinitionTooltip
+          data-testid="test"
+          definition={definition}
+          className="tooltip-class">
+          URL
+        </DefinitionTooltip>
+      );
+
+      const content = screen.getByText(definition);
```
Comment (preetibansalui) [inline packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js]: same here.

**Comment by guidari** [comment]
> How do we check the test coverage percentage locally ?

Hey @2nikhiltom 
Just run this command to get the percentage locally:
`yarn test packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js --coverage --collectCoverageFrom=packages/react/src/components/Tooltip/DefinitionTooltip.tsx`

We left this tips in the comment [here](https://github.com/carbon-design-system/carbon/issues/16319#issuecomment-2346817818)

**Comment by guidari** [comment]
> Does this mean its 100% now? Just want to make sure I'm looking at the right thing. Screenshot 2024-09-12 at 3 12 

@alisonjoseph Yeah that's is correct!
Here it is the [previous percentage vs the [new one](https://app.codecov.io/gh/carbon-design-system/carbon/pull/17434/indirect-changes) you just screen shoted

File: packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ import userEvent from '@testing-library/user-event';
 import React from 'react';
 import { DefinitionTooltip } from '../DefinitionTooltip';
 
-describe('DefintiionTooltip', () => {
+describe('DefinitionTooltip', () => {
   it('should display onClick a defintion provided via prop', async () => {
```
Comment (alisonjoseph) [inline packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js]:
```suggestion
  it('should display onClick a definition provided via prop', async () => {
```

File: packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js (hunk 1 above)
Comment (alisonjoseph) [inline packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js]:
```suggestion
```
agree I think this line can be removed

File: packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js (hunk 2 above)
Comment (alisonjoseph) [inline packages/react/src/components/Tooltip/__tests__/DefinitionTooltip-test.js]:
```suggestion
```

**Review by alisonjoseph — APPROVED** [review]
couple tiny cleanup things but test coverage looks good 💯

**Review by guidari — APPROVED** [review]
Looks good after the clean up from Alison's comment

**Comment by tay1orjones** [comment]
Updated, thanks everyone!

**Pull request merged**